### PR TITLE
Strumenta#66 Printing AST skips private list fields

### PR DESCRIPTION
**Pull request by loradd**

### CODE REVIEW

File: core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt [hunk 1]
```diff
@@ -50,34 +50,36 @@ fun Node.debugPrint(indent: String = "", configuration: DebugPrintConfiguration
             } else {
                 val mt = property.returnType.javaType
                 if (mt is ParameterizedType && mt.rawType == List::class.java) {
-                    if (property.get(this) == null) {
-                        sb.append("$indent$indentBlock${property.name} = null")
-                    } else {
-                        val value = property.get(this) as List<*>
-                        if (value.isEmpty() && configuration.skipEmptyCollections) {
-                            // nothing to do
+                    if (property.visibility == PUBLIC) {
```
Comment (ftomassetti) [inline core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt:53]: Would it make sense to consider package-protected properties? If not, should we add a warning or write this somewhere?

**Review by ftomassetti — APPROVED** [review]
+1

File: core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt (hunk 1 above)
Comment (loradd) [reply core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt:53]: What about handling both private, protected and internal properties according to some dedicated configuration parameter in `DebugPrintConfiguration`, e.g. `val skipProtectedProperties: Boolean = false`?

Comment (loradd) [reply core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt:53]: > Would it make sense to consider package-protected properties? If not, should we add a warning or write this somewhere?

What about handling both private, protected and internal properties according to some dedicated configuration parameter in `DebugPrintConfiguration`, e.g. `val skipProtectedProperties: Boolean = false`?

Comment (ftomassetti) [reply core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt:53]: I think that may be useful, yes

**Comment by ftomassetti** [comment]
Thank you Lorenzo. Can you solve the conflicts and merge the PR? I think you should have the rights to do so

**Pull request merged**